docs/install/osx_setup.md

```diff
@@ -100,14 +104,21 @@ After you have installed the dependencies, pull the MXNet source code from Git a
 The file called ```osx.mk``` has the configuration required for building MXNet on OS X. First copy ```make/osx.mk``` into ```config.mk```, which is used by the ```make``` command:
 
 ```bash
-    git clone --recursive https://github.com/apache/incubator-mxnet ~/mxnet
-    cd ~/mxnet
-    cp make/osx.mk ./config.mk
-    echo "USE_BLAS = openblas" >> ./config.mk
-    echo "ADD_CFLAGS += -I/usr/local/opt/openblas/include" >> ./config.mk
-    echo "ADD_LDFLAGS += -L/usr/local/opt/openblas/lib" >> ./config.mk
-    echo "ADD_LDFLAGS += -L/usr/local/lib/graphviz/" >> ./config.mk
-    make -j$(sysctl -n hw.ncpu)
+git clone --recursive https://github.com/apache/incubator-mxnet ~/mxnet
+cd ~/mxnet
+mkdir build && cd build
+cmake \
+    -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
```

Comment:
While `ccache` is really tool for speeding up the compilation time, I still feel we should just provide the minimum required steps for users to build mxnet. Can we suggest these performance tools in a separate doc/section?

Reply:
Got error `CMake Error: The source directory "/Users/lnyuan/work/mxnet/build" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.` following this instruction.

Reply:
I'm all for speed. I think we should include it. 
Unless there's a lot of complexity added... does the user have to setup ccache or is it a seamless installation during the dependencies setup?

Reply:
If that's the case, we should make this default rather than letting users to specify. I have found this instruction too complicated to use for me. Copy and paste multiple lines does not work well.

Reply:
These should be bare minimum steps. Requiring ccache is an advanced thing and I'd prefer to not have it in the basic instructions.

Reply:
Agree. ccache is not a must and it's an advanced compiler tool that should be up to users to choose.
To @aaronmarkham 's question, yes it does require user to set up `ccache` and it's not seamless installation.